tests/attributes/src/lib.rs

```diff
@@ -180,3 +180,46 @@ unfold instance impl: Core.Ops.Arith.t_Sub $:Int $:Int =
         }
     }
 }
+
+/// Illustration of the `refinement_type` macro that helps creating refinement types via thin newtype wrappers.
+mod refinement_types {
+    use hax_lib::*;
+
+    #[hax_lib::refinement_type(|x| x >= MIN && x <= MAX)]
+    pub struct BoundedU8<const MIN: u8, const MAX: u8>(u8);
+
+    pub fn bounded_u8(x: BoundedU8<12, 15>, y: BoundedU8<10, 11>) -> BoundedU8<1, 23> {
+        BoundedU8::new(x.get() + y.get())
```

Comment:
This is pretty ugly. Do we really need a `get` here?
When we talked about this, we said we wanted as little clutter as possible. That's not the case here. In our example this was
```rust
(x + y).into()
```

Reply:
Maybe I should just kill this example all together: this PR is really just about refinements. So I don't include any Rust libraries or facilities to not to have this kinds of things
The impls for easy addition multiplication etc are implemented in the PR about bounded integers
Here is just a test / examples